include/hip/hcc_detail/hip_complex.h

```diff
@@ -24,7 +24,7 @@ THE SOFTWARE.
 #define HIP_INCLUDE_HIP_HCC_DETAIL_HIP_COMPLEX_H
 
 #include "hip/hcc_detail/hip_vector_types.h"
-#include <math.h>
+#include <cmath>
```

Comment:
@yxsamliu I'm not sure this is correct - math is only available in C++, so it'd seem more reasonable to put it under the __cplusplus guard, methinks?

Reply:
That should be fine. What is the issue with HIP-Clang? There should only be a __device__ exp in HIP, which should not clash (in C++).

Reply:
I am still investigating whether this is a clang bug. Basically, if we have exp defined in std namespace, and we have using std::exp in default name space, a device function calling exp will call this std::exp instead of device exp. It is supposed to check the device attribute but it does not do that.

Probably I should put a "//TODO: Fix bug about device function calling std::exp" there and remove this after the bug is fixed.